scripts/root-ca-retriever/public-certs.ts

```diff
@@ -0,0 +1,414 @@
+
+  return new Promise<MozillaCSVRecord[]>((resolve, reject) => {
+    https
+      .get(MOZILLA_CSV_URL, response => {
+        if (response.statusCode !== 200) {
+          reject(new Error(`Failed to get Mozilla CSV data. Status code: ${response.statusCode}`));
+          return;
+        }
+
+        let data = "";
+        response.on("data", chunk => {
+          data += chunk;
+        });
+
+        response.on("end", () => {
+          try {
+            const records = parseCSV(data, {
+              columns: true,
+              skip_empty_lines: true,
+              trim: true,
+            }) as MozillaCSVRecord[];
+
+            console.log(`Downloaded ${records.length} Mozilla CA records`);
+            resolve(records);
+          } catch (error) {
+            reject(new Error(`Failed to parse CSV: ${error}`));
+          }
+        });
+      })
+      .on("error", err => {
+        reject(err);
+      });
+  });
+}
+
+/**
+ * Extracts individual certificates from a PEM bundle along with their preceding comments.
+ * The certificate name is extracted from the comment block that precedes each certificate.
+ * @param pemContent - The raw PEM bundle content containing multiple certificates
+ * @returns Array of objects containing certificate name and PEM content
+ */
+export function extractCertificatesFromPEM(
+  pemContent: string,
+): { name: string; content: string }[] {
+  const certRegex = /(.*?)\n=+\n(-----BEGIN CERTIFICATE-----[\s\S]*?-----END CERTIFICATE-----)/g;
+  const matches = [];
+  let match;
+
+  while ((match = certRegex.exec(pemContent)) !== null) {
+    const comment = match[1].trim();
+    const certContent = match[2];
+
+    // Extract certificate name from comment - it's usually the last line of the comment
+    const commentLines = comment.split("\n").filter(line => line.trim().length > 0);
+    const name = commentLines[commentLines.length - 1].trim();
+
+    matches.push({
+      name,
+      content: certContent,
+    });
+  }
+
+  return matches;
+}
+
+/**
+ * Creates an inventory of public CA certificates, validating each certificate's format
+ * using Node.js crypto library and converting to PublicCACert objects
+ * @param certsWithNames - Array of certificates with extracted names and PEM content
+ * @returns Array of validated PublicCACert objects
+ * @throws {Error} When certificate validation fails
+ */
+export function inventoryPublicCACertificates(
+  certsWithNames: { name: string; content: string }[],
+): PublicCACert[] {
+  return certsWithNames.map(({ name, content }) => {
+    // Validate certificate format
+    try {
+      new crypto.X509Certificate(content);
+    } catch (error) {
+      throw new Error(
+        `Invalid certificate ${name}: ${error instanceof Error ? error.message : String(error)}`,
+      );
+    }
+
+    return {
+      commonName: name,
+      content,
+    };
+  });
+}
+
+/**
+ * Enriches PublicCACert objects with metadata from Mozilla's CCDB CSV data.
```

Comment:
Should this be `CCADB`?
```suggestion
 * Enriches PublicCACert objects with metadata from Mozilla's CCADB CSV data.
```

Reply:
fixed